src/Microsoft.AspNet.Mvc.Core/Controllers/DefaultControllerActionArgumentBinder.cs

```diff
@@ -64,6 +64,28 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
                         nameof(actionContext));
             }
 
+            if (actionDescriptor.BoundProperties.Count == 0 &&
+                actionDescriptor.Parameters.Count == 0)
+            {
+                return Task.FromResult<IDictionary<string, object>>(
+                    new Dictionary<string, object>(StringComparer.Ordinal));
```

Comment:
I also considered pulling this check out into the invoker, but it seems like a good thing to have the binder at least be called - for extensibility's sake. The code is less artificial if I put the `if` in the invoker, but that seemed like a bad choice for future proofing.